src/compiler/crystal/compiler.cr

```diff
@@ -314,22 +314,29 @@ module Crystal
 
       target_machine.emit_obj_to_file llvm_mod, object_name
 
-      stdout.puts linker_command(program, object_name, output_filename, nil)
+      stdout.puts linker_command(program, [object_name], output_filename, nil)
     end
 
-    private def linker_command(program : Program, object_name, output_filename, output_dir)
+    private def linker_command(program : Program, object_names, output_filename, output_dir)
       if program.has_flag? "windows"
-        if object_name
-          object_name = %("#{object_name}")
-        else
-          object_name = %(%*)
-        end
-
         if link_flags = @link_flags.presence
           link_flags = "/link #{link_flags}"
         end
 
-        %(#{CL} #{object_name} "/Fe#{output_filename}" #{program.lib_flags} #{link_flags})
+        args = %(#{object_names.join(" ")} "/Fe#{output_filename}" #{program.lib_flags} #{link_flags})
+        cmd = "#{CL} #{args}"
+
+        if cmd.to_utf16.size > 32000
+          # The command line would be too big, pass the args through a UTF-16-encoded file instead.
+          # TODO: Do this the proper way when iconv is supported.
+          cmd_filename = "#{output_dir}/link_command.cmd"
+          cmd_16 = "\ufeff#{args}".to_utf16
```

Comment:
What is the `\ufeff`?

Reply:
[Byte order mark](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-16)

Reply:
[BOM](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-16)  :)

Reply:
Can we get a comment explaining this?